galsim/phase_psf.py

```diff
@@ -0,0 +1,1017 @@
+# Copyright (c) 2012-2015 by the GalSim developers team on GitHub
+# https://github.com/GalSim-developers
+#
+# This file is part of GalSim: The modular galaxy image simulation toolkit.
+# https://github.com/GalSim-developers/GalSim
+#
+# GalSim is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+"""@file phase_psf.py
+Utilities for creating PSFs from phase screens.  Essentially evaluates the Fourier optics
+diffraction equation:
+
+PSF(x, y) = int( |FT(aperture(u, v) * exp(i * phase(u, v, x, y, t)))|^2, dt)
+
+where x, y are focal plane coordinates and u, v are pupil plane coordinates
+
+The main classes of note are:
+
+FrozenAtmosphericScreen
+  Class implementing phase(u, v, x, y, t) for frozen flow von Karman type turbulence
+
+ARAtmosphericScreen
```

Comment:
>The prefix AR is completely obtuse to me. 

I'm fine with renaming this.

The reason I had two classes was that FrozenAtmosphericScreen is implemented differently than ARAtmosphericScreen with alpha_mag=1, even though they do represent the same screen (I use this for a unit test, in fact).  ARAtmosphericScreen propagates the screen with the wind by multiplying the phase in Fourier space and then transforming.  FrozenAtmosphericScreen propagates the aperture in the opposite direction instead of propagating the screen, which then avoids a Fourier transform.  I agree that we could hide this from the user though, and still enable both implementations for the unit test with a private kwarg to force the slower Fourier approach.